File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        if(n == 0) return {0};
        vector<int> ans(n+1, -1);
        ans[0] = 0;
        for(int i = 1; i <= n; i++){
            int sum = 0;
            sum += i % 2;
            sum += ans[i / 2];
            ans[i] = sum;
        }
        return ans;
    }
};

File: 0338-counting-bits/0338-counting-bits.js
/**
 * @param {number} n
 * @return {number[]}
 */
var countBits = function(n) {
    if(n === 0) return [0];
    const ans = Array.from({length: n+1}, () => -1);
    ans[0] = 0;

    for(let i = 0; i <= n; i++){
        let sum = 0;
        sum += i%2;
        const leftShift = Math.floor(i / 2);
        sum += ans[leftShift];
        ans[i] = sum;   
    }
    return ans;
};